src/operations_protobuf/convert_attest_key.rs

```diff
@@ -191,6 +289,33 @@ mod test {
         let op_attesting_key_name = String::from("attesting key name");
         let op_credential_blob = vec![0xaa; 32];
         let op_secret = vec![0x11; 16];
+        let op_nonce = vec![0x88; 8];
+
+        let op = Operation::CertifyAndQuote {
+            attested_key_name: op_attested_key_name.clone(),
+            attesting_key_name: Some(op_attesting_key_name.clone()),
+            nonce: op_nonce.clone().into(),
+        };
+
+        let proto: OperationProto = op.try_into().expect("Failed conversion");
+        if let OperationProto {
+            attested_key_name,
+            attesting_key_name,
+            parameters:
+                Some(AttestationMechanismParams {
+                    mechanism:
+                        Some(attestation_mechanism_params::Mechanism::CertifyAndQuote(
+                            attestation_mechanism_params::CertifyAndQuote {
+                                nonce,
+                            },
+                        )),
+                }),
+        } = proto
+        {
+            assert_eq!(attested_key_name, op_attested_key_name);
+            assert_eq!(attesting_key_name, op_attesting_key_name);
+            assert_eq!(nonce, op_nonce);
+        }
```

Comment:
I think we should have an `else` here, in case the conversion produced the wrong thing - if you don't actually get what you expect in that `if let`, this branch is never taken and the test currently passes, but it should actually fail

Reply:
Oh, and you should also fix the existing tests for this! (just the ones in this file, don't know if other files are affected)

Reply:
I've done this but left `panic!("'if let' branch not taken")` in the else statement. This response probably needs to be better but it should be a reasonable placeholder.